### ISSUE TITLE
Microphone permissions cannot be granted but are shown as granted in some UI elements

### ISSUE THREAD

**Issue by nicoroeser** [Eloston/ungoogled-chromium#1070]
## Summary
I am testing with `ungoogled-chromium-81.0.4044.138-1.focal1` from <https://download.opensuse.org/repositories/home:/ungoogled_chromium/Ubuntu_Focal> on Ubuntu 20.04 LTS (Focal Fossa).

The microphone permission does not work as expected: it can never effectively be granted, and if the user grants it for a website, then it shows as granted in one place and as not granted in another.

## Steps to reproduce
These steps work on my test system:
1. Open a website which uses the microphone (I used [our installation](https://meet.ur.de/) of [Jitsi Meet](https://github.com/jitsi/jitsi-meet/) for testing),
2. click on the camera icon with the red „forbidden“ symbol on it in the right part of the address bar,
3. choose to allow microphone access and confirm.

### Actual result
Nothing changes.

### Expected result
Microphone permissions are granted to the website.

## Details
I have tested as well with all extensions turned off. This does not make a difference.

### Inconsistent display and behavior of property settings
1. I open another website that uses the microphone.
2. I click on the icon (normally, on HTTPS pages: lock symbol) in the left part of the address bar. A popup with permissions opens. In this popup, I see “Microphone: Allow”.
3. When I select “Site settings” at the bottom of that popup window, the list appears again, in another browser tab (URI is `chrome://settings/content/siteDetails?site=https%3A%2F%2Fsite.name.example`).
4. In that new tab, it says: “Microphone: Block”. When I click on the dropdown field and select “Allow”, it **does not** change from “Block” to “Allow”.

### Test page
I have created a test page:
```html
<!DOCTYPE html>
<html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
<title>Mic Test</title>
</head>
<body>
<h1>Mic Test</h1>
<p>Testing…</p>
<div id="result">Result will appear here.</div>
<script type="application/javascript">/*<![CDATA[*/
"use strict";

var div = document.getElementById("result");

navigator.mediaDevices.getUserMedia({audio: true, video: false}).then(function(a) {
    div.innerHTML = "SUCCESS: " + a;
}).catch(function(e) {
    div.innerHTML = "ERROR: " + e;
}).then(function() {
    div.innerHTML += '<br/>';
    navigator.permissions.query({name:'microphone'}).then(function(permStatus) {
        div.innerHTML += 'permission state=' + permStatus.state;
    }).catch(function(err) {
        div.innerHTML += 'error: ' + err;
    });
});
/*]]>*/</script>
</body>
</html>
```

When I open that test page using a `file:` URI, the same problem appears. The produced `div` element in the page is:
```html
<div id="result">ERROR: NotAllowedError: Permission denied<br>permission state=denied</div>
```

I am unable to change the microphone permissions for that test page, the status always stays blocked.

By the way, this looks like #600 the other way round.

If you have some pointers for further debugging, I’ll happily do further testing.

**Comment by Eloston** [comment]
Is it because of  https://ungoogled-software.github.io/ungoogled-chromium-wiki/faq#why-is-my-microphone-not-working?

This looks like a Debian-specific change in the master preferences; I might see to reverting this change since it doesn't seem very useful (and quite inconvenient).

**Comment by nicoroeser** [comment]
> Is it because of https://ungoogled-software.github.io/ungoogled-chromium-wiki/faq#why-is-my-microphone-not-working?

Ah – yes, it is! The first part of the answer applies:
> 1. You set your preferences or the master preferences with `"audio_capture_enabled": false`.

Many thanks for the pointer! :smiley: Sorry, I hadn’t noticed the FAQ document earlier.

> 
> This looks like a Debian-specific change in the master preferences; I might see to reverting this change since it doesn't seem very useful (and quite inconvenient). EDIT: Reverted in [ungoogled-software/ungoogled-chromium-debian@1da7167](https://github.com/ungoogled-software/ungoogled-chromium-debian/commit/1da716756ef105fc5e2656c6171862f03f8962f7)

Looks good to me, thanks.

So even though my initial problem is solved, I still see three things that can be done/improved:

1. when `audio_capture_enabled` is set to `false`, the UI should not behave in the way that I have described in [my initial comment](#issue-635179373) under “Inconsistent display and behavior of property settings”. The user should receive some indication in the UI which explains that the permission setting cannot be changed due to `audio_capture_enabled` being disabled (or that it will be ignored). And both locations in the UI which I mentioned should show a consistent state and behave the same way, or users will be confused.
   I don’t know whether this is an issue for _this_ project or should rather be handled upstream.
1. I have not been able to find the UI for changing `audio_capture_enabled`. Perhaps I just don’t understand the preferences page (`chrome://settings/`). If there really is no UI for this, a small addition might be handy.
1. (more out of curiosity, not important:) where did the setting in `master_preferences` originate, and for what reason has it been set?

**Comment by Eloston** [comment]
> (more out of curiosity, not important:) where did the setting in master_preferences originate, and for what reason has it been set?

I think I set it a long time ago when I created the Debian packaging. I didn't really know the consequences of the flag, nor did I realize how obscure its effects are. No other packaging of ungoogled-chromium sets this AFAIK, so this shouldn't really be an issue in the future.

> I have not been able to find the UI for changing audio_capture_enabled. Perhaps I just don’t understand the preferences page (chrome://settings/). If there really is no UI for this, a small addition might be handy.

I don't think anyone is supposed to set the flag, unless you're an administrator with a specific use-case where the microphone should never be used. Whatever the real use-case is, I don't think anyone that uses ungoogled-chromium will run into the issue unless they're manually adding those flags themselves.

If I am wrong about this flag, then I'd be happy to consider your UI suggestions.
